cmd/snap-update-ns/bootstrap.c

```diff
@@ -368,6 +368,40 @@ void process_arguments(int argc, char *const *argv, const char **snap_name_out,
 				// Processing the user-fstab file implies we're being
 				// called from snap-confine.
 				should_setns = false;
+			} else if (!strcmp(arg, "-u")) {
+				const char *uid_text = argv[i + 1];
+				if (i + 1 == argc || uid_text == NULL) {
+					bootstrap_msg = "-u requires an argument";
+					bootstrap_errno = 0;
+					return;
+				}
+				errno = 0;
+				char *uid_text_end = NULL;
+				int parsed_uid = (int)strtol(uid_text, &uid_text_end, 10);
+				if ((parsed_uid == 0 && errno != 0)
+						|| (*uid_text == '\0')
+						|| (*uid_text != '\0' && uid_text_end != NULL
+							&& *uid_text_end != '\0')) {
+					bootstrap_msg = "cannot parse user id";
+					bootstrap_errno = errno;
+					return;
+				}
+				if (parsed_uid < 0) {
+					bootstrap_msg = "user id cannot be negative";
+					bootstrap_errno = 0;
+					return;
+				}
```

Comment:
IMHO this parsing code could be extracted into a helper, resulting in more compact body of process_arguments.